packages/pieces/xero/src/lib/actions/create-invoice.ts

```diff
@@ -0,0 +1,203 @@
+        "SentToContact": false,
+        "CurrencyRate": 1,
+        "IsDiscounted": false,
+        "HasErrors": false,
+        "InvoicePaymentServices": [],
+        "Contact": {
+          "ContactID": "e595b89d-347e-447d-a859-ee03340ec88d",
+          "ContactStatus": "ACTIVE",
+          "Name": "DSAP",
+          "EmailAddress": "dq@example.com",
+          "BankAccountDetails": "",
+          "Addresses": [
+            {
+              "AddressType": "STREET",
+              "City": "",
+              "Region": "",
+              "PostalCode": "",
+              "Country": ""
+            },
+            {
+              "AddressType": "POBOX",
+              "City": "",
+              "Region": "",
+              "PostalCode": "",
+              "Country": ""
+            }
+          ],
+          "Phones": [
+            {
+              "PhoneType": "DEFAULT",
+              "PhoneNumber": "",
+              "PhoneAreaCode": "",
+              "PhoneCountryCode": ""
+            },
+            {
+              "PhoneType": "DDI",
+              "PhoneNumber": "",
+              "PhoneAreaCode": "",
+              "PhoneCountryCode": ""
+            },
+            {
+              "PhoneType": "FAX",
+              "PhoneNumber": "",
+              "PhoneAreaCode": "",
+              "PhoneCountryCode": ""
+            },
+            {
+              "PhoneType": "MOBILE",
+              "PhoneNumber": "",
+              "PhoneAreaCode": "",
+              "PhoneCountryCode": ""
+            }
+          ],
+          "UpdatedDateUTC": "/Date(1679982402187+0000)/",
+          "ContactGroups": [],
+          "IsSupplier": false,
+          "IsCustomer": true,
+          "SalesTrackingCategories": [],
+          "PurchasesTrackingCategories": [],
+          "ContactPersons": [],
+          "HasValidationErrors": false
+        },
+        "DateString": "2023-03-04T00:00:00",
+        "Date": "/Date(1677888000000+0000)/",
+        "DueDateString": "2023-03-18T00:00:00",
+        "DueDate": "/Date(1679097600000+0000)/",
+        "BrandingThemeID": "4b3fdee3-c068-4121-a348-db3b4b7934d9",
+        "Status": "SUBMITTED",
+        "LineAmountTypes": "Exclusive",
+        "LineItems": [
+          {
+            "Description": "Description",
+            "UnitAmount": 200,
+            "TaxType": "NONE",
+            "TaxAmount": 0,
+            "LineAmount": 200,
+            "AccountCode": "200",
+            "Tracking": [],
+            "Quantity": 1,
+            "LineItemID": "43331784-c61d-4a56-8488-312b76f5c705",
+            "ValidationErrors": []
+          }
+        ],
+        "SubTotal": 200,
+        "TotalTax": 0,
+        "Total": 200,
+        "UpdatedDateUTC": "/Date(1679982403190+0000)/",
+        "CurrencyCode": "KES"
+      }
+    ]
+  },
+  props: {
+    authentication: props.authentication,
+    tenant_id: props.tenant_id,
+    invoice_id: props.invoice_id,
+    contact_id: props.contact_id(true),
```

Comment:
Let's replace contact_id with contact_name (required) and contact_email (optional).